WooCommerce/Classes/POS/ViewModels/ItemListViewModel.swift

```diff
@@ -42,25 +27,135 @@ final class ItemListViewModel: ObservableObject {
         do {
             state = .loading
             items = try await itemProvider.providePointOfSaleItems()
-            state = .loaded
+            if items.count == 0 {
+                let itemListEmpty = EmptyModel(title: Constants.emptyProductsTitle,
+                                                  subtitle: Constants.emptyProductsSubtitle,
+                                                  hint: Constants.emptyProductsHint,
+                                                  buttonText: Constants.emptyProductsButtonTitle)
+                state = .empty(itemListEmpty)
+            } else {
+                state = .loaded(items)
+            }
         } catch {
             DDLogError("Error on load while fetching product data: \(error)")
-            state = .error
+            let itemListError = ErrorModel(title: Constants.failedToLoadTitle,
+                                      subtitle: Constants.failedToLoadSubtitle,
+                                      buttonText: Constants.failedToLoadButtonTitle)
+            state = .error(itemListError)
         }
     }
 
     @MainActor
     func reload() async {
         do {
+            // TODO:
+            // Resolve duplication with populatePointOfSaleItems()
```

Comment:
I'll resolve this duplication separately, since requires a bit more testing around updating products via the web/app and confirming that the reload works as expected.

Reply:
> Out of curiosity, why removing one of reload() or populatePointOfSaleItems() could cause problems? They look to be doing the same thing or am I missing something?

Good question, I had to rethink this again 🤔 Right now the only difference is that we clear the loaded items on `reload()`, and we do not do it on `populatePointOfSaleItems()`  because it's only called once on model initialization, so it doesn't need clearing. 

I think this difference was more important previously, when we were fetching products from storage rather than from the network as we do now, since could be discrepancies between what was stored and what was fetched, however, now there shouldn't be much difference since in both cases we would have a new fresh list of the latest 100 simple products from the remote. 

I'll keep it for a different PR just so we don't mix changes on this one and can improve the tests as well, but I'd say that right now they're doing effectively the same thing.

Reply:
Logged here: https://github.com/woocommerce/woocommerce-ios/issues/13307